copy/copy.go

```diff
@@ -237,15 +243,23 @@ func Image(policyContext *signature.PolicyContext, destRef, srcRef types.ImageRe
 
 // copyLayers copies layers from src/rawSource to dest, using and updating ic.manifestUpdates if necessary and ic.canModifyManifest.
 func (ic *imageCopier) copyLayers() error {
+	origSrcInfos := ic.src.LayerInfos()
 	srcInfos := ic.src.LayerInfos()
+
+	if ic.layerEditor != nil {
+		srcInfos = ic.layerEditor(ic.src.LayerInfos())
```

Comment:
Calling LayerInfos three times is a bit much… at the very least `srcInfos = ic.layerEditor(srcInfos)`